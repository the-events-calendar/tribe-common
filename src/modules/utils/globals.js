/**
 * @todo: handle globals in a better way
 */

export const get = ( key, defaultValue ) => window[ key ] || defaultValue;
export const google = () => get( 'google' );
export const wpApi = wp.api;
export const wpApiRequest = wp.apiRequest;
export const wpComponents = wp.components;
export const wpData = wp.data;
export const wpEditor = wp.blockEditor || wp.editor;
export const wpHooks = wp.hooks;

export const wpDataSelectCoreEditor = () => wpData.select( 'core/block-editor' ) ||
	wpData.select( 'core/editor' );

export const IconButton = wpComponents.Button || wpComponents.IconButton;

// Localized Config
export const config = () => get( 'tribe_editor_config', {} );

// Common
export const common = () => config().common || {};
export const adminUrl = () => common().adminUrl || '';
export const rest = () => common().rest || {};
export const restNonce = () => rest().nonce || {};
export const dateSettings = () => common().dateSettings || {};
export const editorConstants = () => common().constants || {};
export const list = () => ( {
	countries: common().countries || {},
	us_states: common().usStates || {},
} );

// TEC
export const tec = () => config().events || {};
export const editor = () => tec().editor || {};
export const settings = () => tec().settings || {};
export const mapsAPI = () => tec().googleMap || {};
export const priceSettings = () => tec().priceSettings || {};
export const tecDateSettings = () => tec().dateSettings || {};
export const timezoneHtml = () => tec().timezoneHTML || '';
export const defaultTimes = () => tec().defaultTimes || {};
export const timezone = () => tec().timeZone || {};

// PRO
export const pro = () => config().eventsPRO || {};
export const editorDefaults = () => pro().defaults || {};

// Tickets
export const tickets = () => config().tickets || {};

// Tickets Plus
export const ticketsPlus = () => config().ticketsPlus || {};
export const iacVars = () => ticketsPlus().iacVars || {};

// Post Objects
export const postObjects = () => config().post_objects || {};
export const post = () => config().post || {};

// Blocks
export const blocks = () => config().blocks || {};
