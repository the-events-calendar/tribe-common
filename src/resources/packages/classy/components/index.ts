export { CenteredSpinner } from './CenteredSpinner';
export { CurrencyInput } from './CurrencyInput';
export { CurrencySelector } from './CurrencySelector';
export { DatePicker } from './DatePicker';
export { EndSelector } from './EndSelector';
export { ErrorBoundary } from './ErrorBoundary';
export * from './Icons';
export { LabeledInput } from './LabeledInput';
export { TimePicker } from './TimePicker';
export { TimeZone } from './TimeZone';
export { TinyMceEditor } from './TinyMceEditor';
export { StartSelector } from './StartSelector';

// Note: the Classy application is not exported by design, only components are.
