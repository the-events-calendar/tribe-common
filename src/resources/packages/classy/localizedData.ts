import { LocalizedData } from './types/LocalizedData';
import { WPDataRegistry } from '@wordpress/data/build-types/registry';

declare global {
	interface Window {
		tec: {
			common: {
				classy: {
					data: LocalizedData;
					registry: WPDataRegistry;
				};
			};
		};
	}
}

export const localizedData: LocalizedData = window?.tec?.common?.classy?.data ?? {
	settings: {
		defaultCurrency: {
			code: 'USD',
			symbol: '$',
			position: 'prefix',
		},
		timezoneString: 'UTC',
		timezoneChoice: '',
		startOfWeek: 0,
		endOfDayCutoff: {
			hours: 0,
			minutes: 0,
		},
		dateWithYearFormat: 'F j, Y',
		dateWithoutYearFormat: 'F j',
		monthAndYearFormat: 'F Y',
		compactDateFormat: 'n/j/Y',
		dataTimeSeparator: ' @ ',
		timeRangeSeparator: ' - ',
		timeFormat: 'g:i A',
		timeInterval: 15,
	},
};
