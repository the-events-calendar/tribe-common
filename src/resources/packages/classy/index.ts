import { whenEditorIsReady } from './functions/whenEditorIsReady';
import { hideInserterToggle, hideZoomOutButton, hideSidebarBlockTab } from './functions/editorModifications';
import {
	initApp as initClassyApp,
	insertElement as insertClassyElement,
} from './functions/classy';
import { getLocalizedData, getSettings } from './localizedData';
import './style.pcss';

whenEditorIsReady().then( () => {
	hideZoomOutButton();
	hideInserterToggle();
	hideSidebarBlockTab();
	initClassyApp();
	insertClassyElement();
} );

// Re-exports that will appear under `window.tec.common.classy.<re-export>`.
export * as components from './components';
export * as fields from './fields';
export * as functions from './functions';
export * as store from './store';

/*
 * Re-export localized data accessors and not the localized data object directly.
 * Packages outside of this will be able to access the localized data in one of two ways:
 *
 * Recommended:
 * - import {getLocalizedData, getSettings} from '@tec/common/classy/localizedData';
 *
 * Not ideal but still possible:
 * - const {getLocalizedData, getSettings} = window.tec.common.classy.localizedData;
 */
export const localizedData = {
	getLocalizedData,
	getSettings,
};
