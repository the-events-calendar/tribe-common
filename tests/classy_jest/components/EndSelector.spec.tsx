import * as React from 'react';
import { render } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { afterEach, beforeEach, describe, expect, it, jest } from '@jest/globals';
import { EndSelector } from '../../../src/resources/packages/classy/components';
import { keyDownEscape } from '../_support/userEvents';
import { getDefault as getDefaultLocalizedData } from '../../../src/resources/packages/classy/localizedData';
import { LocalizedData } from '@tec/common/classy/types/LocalizedData';
import TestProvider from '../_support/TestProvider';

// Save the original localized data here.
let originalLocalizedData: LocalizedData;
const timePickerSelector = '.classy-field__input--end-time .components-combobox-control__input';
const datePickerButton = '.classy-field__control--date-picker input.components-input-control__input';

describe( 'EndSelector Component', () => {
	const defaultProps = {
		dateWithYearFormat: 'Y-m-d',
		endDate: new Date( '2023-12-23 13:00:00' ),
		highlightTime: false,
		isAllDay: false,
		isMultiday: false,
		isSelectingDate: false as const,
		onChange: jest.fn(),
		onClick: jest.fn(),
		onClose: jest.fn(),
		startDate: new Date( '2023-12-23 10:00:00' ),
		startOfWeek: 0 as const,
		timeFormat: 'g:i a',
	};

	beforeEach( () => {
		// Set the localized data to a known state.
		originalLocalizedData = window.tec.common.classy.data;
		window.tec.common.classy.data = getDefaultLocalizedData();
	} );

	afterEach( () => {
		// Clean up.
		if ( originalLocalizedData ) {
			window.tec.common.classy.data = originalLocalizedData;
		}
	} );

	it( 'renders correctly with default props', () => {
		const { container } = render(
			<TestProvider>
				<EndSelector { ...defaultProps } />
			</TestProvider>
		);

		expect( container.firstChild ).toMatchSnapshot();
	} );

	describe( 'all day events', () => {
		it( 'renders "All Day" label when isAllDay is true', () => {
			const props = {
				...defaultProps,
				isAllDay: true,
			};

			const { container, getByText, queryByText } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( getByText( 'All Day' ) ).toBeTruthy();
			expect( queryByText( 'End Time' ) ).toBeNull();
			expect( container.querySelector( '.classy-field__input--end-time' ) ).toBeNull();
			expect( container.firstChild ).toMatchSnapshot();
		} );

		it( 'renders with time picker when isAllDay is false', () => {
			const props = {
				...defaultProps,
				isAllDay: false,
			};

			const { container, getByText } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( getByText( 'End Time' ) ).toBeTruthy();
			expect( container.querySelector( '.classy-field__input--end-time' ) ).toBeTruthy();
			expect( container.firstChild ).toMatchSnapshot();
		} );
	} );

	describe( 'multiday events', () => {
		it( 'renders date separator and date picker when isMultiday is true', () => {
			const props = {
				...defaultProps,
				isMultiday: true,
				endDate: new Date( '2023-12-25 13:00:00' ),
			};

			const { container, getByText } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( getByText( 'to' ) ).toBeTruthy();
			expect( getByText( 'Date' ) ).toBeTruthy();
			expect( container.querySelector( '.classy-field__separator--dates' ) ).toBeTruthy();
			expect( container.querySelector( '.classy-field__input--start-date' ) ).toBeTruthy();
			expect( container.firstChild ).toMatchSnapshot();
		} );

		it( 'does not render date separator when isMultiday is false', () => {
			const props = {
				...defaultProps,
				isMultiday: false,
			};

			const { queryByText } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( queryByText( 'to' ) ).toBeNull();
			expect( queryByText( 'Date' ) ).toBeNull();
		} );

		it( 'passes null as startDate to TimePicker when isMultiday is true', () => {
			const props = {
				...defaultProps,
				isMultiday: true,
				isAllDay: false,
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			// TimePicker should be rendered with null startDate for multiday events
			const timePicker = container.querySelector( '.classy-field__input--end-time' );
			expect( timePicker ).toBeTruthy();
		} );

		it( 'passes startDate to TimePicker when isMultiday is false', () => {
			const props = {
				...defaultProps,
				isMultiday: false,
				isAllDay: false,
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			// TimePicker should be rendered with startDate for single-day events
			const timePicker = container.querySelector( '.classy-field__input--end-time' );
			expect( timePicker ).toBeTruthy();
		} );
	} );
	//
	describe( 'date selection', () => {
		it( 'shows DatePicker popover when isSelectingDate is endDate and isMultiday', () => {
			const props = {
				...defaultProps,
				isMultiday: true,
				isSelectingDate: 'endDate' as const,
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( container.firstChild ).toMatchSnapshot();
		} );

		it( 'does not show DatePicker popover when isSelectingDate is false', () => {
			const props = {
				...defaultProps,
				isMultiday: true,
				isSelectingDate: false as const,
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( container.firstChild ).toMatchSnapshot();
		} );

		it( 'does not show DatePicker popover when isSelectingDate is startDate', () => {
			const props = {
				...defaultProps,
				isMultiday: true,
				isSelectingDate: 'startDate' as const,
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( container.firstChild ).toMatchSnapshot();
		} );
	} );

	describe( 'time selection', () => {
		it( 'calls onChange with correct parameters when time is changed', async () => {
			const user = userEvent.setup();
			const props = {
				...defaultProps,
				isAllDay: false,
				onChange: jest.fn(),
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			// Find the time input (this would be inside the TimePicker component).
			const timeInput = container.querySelector( timePickerSelector );

			expect( timeInput ).not.toBeNull();

			await user.click( timeInput );
			await user.clear( timeInput );
			await user.type( timeInput, '2:30 pm' );
			await user.type( timeInput, '{enter}' );

			// Check that onChange was called with the correct parameters
			expect( props.onChange ).toHaveBeenCalledWith( 'endTime', '2023-12-23 14:30:00' );
		} );

		it( 'highlights time when highlightTime is true', () => {
			const props = {
				...defaultProps,
				highlightTime: true,
				isAllDay: false,
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( container.firstChild ).toMatchSnapshot();
		} );
	} );

	describe( 'event handlers', () => {
		it( 'passes onClick handler to DatePicker', async () => {
			const user = userEvent.setup();
			const onClick = jest.fn();
			const props = {
				...defaultProps,
				isMultiday: true,
				onClick,
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			// The onClick would be triggered through the DatePicker component.
			const dateButton = container.querySelector( datePickerButton );

			expect( dateButton ).not.toBeNull();

			await user.click( dateButton );

			// Check that onClick was called with the expected field
			expect( onClick ).toHaveBeenCalled();
		} );

		it( 'passes onClose handler to DatePicker', async () => {
			const onClose = jest.fn();
			const props = {
				...defaultProps,
				isMultiday: true,
				isSelectingDate: 'endDate' as const,
				onClose,
			};

			render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			// Select the popover element from the document
			const popover = document.body.querySelector( '.classy-component__popover--calendar' );
			if ( popover ) {
				await keyDownEscape( popover );
			}

			// Check that onClose was called
			expect( onClose ).toHaveBeenCalled();
		} );

		it( 'calls onChange with endDate when date changes', async () => {
			const user = userEvent.setup();
			const onChange = jest.fn();
			const props = {
				...defaultProps,
				isMultiday: true,
				isSelectingDate: 'endDate' as const,
				onChange,
			};

			const { getByText } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			// Click on a different date.
			const dateToClick = getByText( '25' );
			if ( dateToClick ) {
				await user.click( dateToClick );
			}

			// Check that onChange was called with the correct parameters
			expect( onChange ).toHaveBeenCalledWith( 'endDate', '2023-12-25T13:00:00' );
		} );

		it( 'calls onChange with endTime when time changes', async () => {
			const user = userEvent.setup();
			const onChange = jest.fn();
			const props = {
				...defaultProps,
				isAllDay: false,
				onChange,
			};

			const { container } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			// Simulate time change through TimePicker
			const timeInput = container.querySelector( timePickerSelector );

			expect( timeInput ).not.toBeNull();

			await user.click( timeInput );
			await user.clear( timeInput );
			await user.type( timeInput, '3:30 pm' );
			await user.type( timeInput, '{enter}' );

			// Check that onChange was called with the correct parameters
			expect( onChange ).toHaveBeenCalledWith( 'endTime', '2023-12-23 15:30:00' );
		} );
	} );

	describe( 'time interval from store', () => {
		it( 'uses time interval from WordPress data store', () => {
			// Set up the localized data with a specific time interval.
			window.tec.common.classy.data.settings.timeInterval = 20;

			const { container } = render(
				<TestProvider>
					<EndSelector { ...defaultProps } />
				</TestProvider>
			);

			expect( container.firstChild ).toMatchSnapshot();
		} );
	} );

	describe( 'accessibility', () => {
		it( 'renders accessible titles for date and time inputs', () => {
			const props = {
				...defaultProps,
				isMultiday: true,
				isAllDay: false,
			};

			const { getByText } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( getByText( 'Date' ) ).toBeTruthy();
			expect( getByText( 'End Time' ) ).toBeTruthy();
		} );

		it( 'renders only End Time title when not multiday', () => {
			const props = {
				...defaultProps,
				isMultiday: false,
				isAllDay: false,
			};

			const { getByText, queryByText } = render(
				<TestProvider>
					<EndSelector { ...props } />
				</TestProvider>
			);

			expect( queryByText( 'Date' ) ).toBeNull();
			expect( getByText( 'End Time' ) ).toBeTruthy();
		} );
	} );
} );
