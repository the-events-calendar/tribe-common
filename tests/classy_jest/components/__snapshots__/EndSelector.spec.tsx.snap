// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndSelector Component all day events renders "All Day" label when isAllDay is true 1`] = `
<span
  class="classy-field__separator classy-field__separator--dates"
>
  All Day
</span>
`;

exports[`EndSelector Component all day events renders with time picker when isAllDay is false 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 13px;
  box-sizing: border-box;
}

.emotion-0 *,
.emotion-0 *::before,
.emotion-0 *::after {
  box-sizing: inherit;
}

.components-panel__row .emotion-2 {
  margin-bottom: inherit;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: calc(4px * 2);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  height: 38px;
  padding-left: calc(4px * 2);
  padding-right: calc(4px * 2);
}

.emotion-5>* {
  min-width: 0;
}

.emotion-7 {
  display: block;
  max-height: 100%;
  max-width: 100%;
  min-height: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  class="classy-field__input classy-field__input--end-time"
>
  <div
    class="classy-field__input-title"
  >
    <h4>
      End Time
    </h4>
  </div>
  <div>
    <div
      class="components-base-control classy-field__control classy-field__control--input classy-field__control--time-picker components-combobox-control emotion-0 emotion-1"
    >
      <div
        class="components-base-control__field emotion-2 emotion-3"
      >
        <div
          class="components-combobox-control__suggestions-container"
          tabindex="-1"
        >
          <div
            class="components-flex emotion-4 emotion-5 emotion-6"
            data-wp-c16t="true"
            data-wp-component="Flex"
          >
            <div
              class="components-flex-item components-flex-block emotion-7 emotion-6"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <input
                aria-autocomplete="list"
                aria-describedby="components-form-token-suggestions-howto-combobox-control-1"
                aria-expanded="false"
                autocomplete="off"
                class="components-combobox-control__input components-form-token-field__input"
                id="components-form-token-input-combobox-control-1"
                role="combobox"
                size="8"
                type="text"
                value="1:00 pm"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EndSelector Component date selection does not show DatePicker popover when isSelectingDate is false 1`] = `
<span
  class="classy-field__separator classy-field__separator--dates"
>
  to
</span>
`;

exports[`EndSelector Component date selection does not show DatePicker popover when isSelectingDate is startDate 1`] = `
<span
  class="classy-field__separator classy-field__separator--dates"
>
  to
</span>
`;

exports[`EndSelector Component date selection shows DatePicker popover when isSelectingDate is endDate and isMultiday 1`] = `
<span
  class="classy-field__separator classy-field__separator--dates"
>
  to
</span>
`;

exports[`EndSelector Component multiday events renders date separator and date picker when isMultiday is true 1`] = `
<span
  class="classy-field__separator classy-field__separator--dates"
>
  to
</span>
`;

exports[`EndSelector Component renders correctly with default props 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 13px;
  box-sizing: border-box;
}

.emotion-0 *,
.emotion-0 *::before,
.emotion-0 *::after {
  box-sizing: inherit;
}

.components-panel__row .emotion-2 {
  margin-bottom: inherit;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: calc(4px * 2);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  height: 38px;
  padding-left: calc(4px * 2);
  padding-right: calc(4px * 2);
}

.emotion-5>* {
  min-width: 0;
}

.emotion-7 {
  display: block;
  max-height: 100%;
  max-width: 100%;
  min-height: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  class="classy-field__input classy-field__input--end-time"
>
  <div
    class="classy-field__input-title"
  >
    <h4>
      End Time
    </h4>
  </div>
  <div>
    <div
      class="components-base-control classy-field__control classy-field__control--input classy-field__control--time-picker components-combobox-control emotion-0 emotion-1"
    >
      <div
        class="components-base-control__field emotion-2 emotion-3"
      >
        <div
          class="components-combobox-control__suggestions-container"
          tabindex="-1"
        >
          <div
            class="components-flex emotion-4 emotion-5 emotion-6"
            data-wp-c16t="true"
            data-wp-component="Flex"
          >
            <div
              class="components-flex-item components-flex-block emotion-7 emotion-6"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <input
                aria-autocomplete="list"
                aria-describedby="components-form-token-suggestions-howto-combobox-control-0"
                aria-expanded="false"
                autocomplete="off"
                class="components-combobox-control__input components-form-token-field__input"
                id="components-form-token-input-combobox-control-0"
                role="combobox"
                size="8"
                type="text"
                value="1:00 pm"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EndSelector Component time interval from store uses time interval from WordPress data store 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 13px;
  box-sizing: border-box;
}

.emotion-0 *,
.emotion-0 *::before,
.emotion-0 *::after {
  box-sizing: inherit;
}

.components-panel__row .emotion-2 {
  margin-bottom: inherit;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: calc(4px * 2);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  height: 38px;
  padding-left: calc(4px * 2);
  padding-right: calc(4px * 2);
}

.emotion-5>* {
  min-width: 0;
}

.emotion-7 {
  display: block;
  max-height: 100%;
  max-width: 100%;
  min-height: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  class="classy-field__input classy-field__input--end-time"
>
  <div
    class="classy-field__input-title"
  >
    <h4>
      End Time
    </h4>
  </div>
  <div>
    <div
      class="components-base-control classy-field__control classy-field__control--input classy-field__control--time-picker components-combobox-control emotion-0 emotion-1"
    >
      <div
        class="components-base-control__field emotion-2 emotion-3"
      >
        <div
          class="components-combobox-control__suggestions-container"
          tabindex="-1"
        >
          <div
            class="components-flex emotion-4 emotion-5 emotion-6"
            data-wp-c16t="true"
            data-wp-component="Flex"
          >
            <div
              class="components-flex-item components-flex-block emotion-7 emotion-6"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <input
                aria-autocomplete="list"
                aria-describedby="components-form-token-suggestions-howto-combobox-control-15"
                aria-expanded="false"
                autocomplete="off"
                class="components-combobox-control__input components-form-token-field__input"
                id="components-form-token-input-combobox-control-15"
                role="combobox"
                size="8"
                type="text"
                value="1:00 pm"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`EndSelector Component time selection highlights time when highlightTime is true 1`] = `
.emotion-0 {
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 13px;
  box-sizing: border-box;
}

.emotion-0 *,
.emotion-0 *::before,
.emotion-0 *::after {
  box-sizing: inherit;
}

.components-panel__row .emotion-2 {
  margin-bottom: inherit;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: calc(4px * 2);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  height: 38px;
  padding-left: calc(4px * 2);
  padding-right: calc(4px * 2);
}

.emotion-5>* {
  min-width: 0;
}

.emotion-7 {
  display: block;
  max-height: 100%;
  max-width: 100%;
  min-height: 0;
  min-width: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  class="classy-field__input classy-field__input--end-time"
>
  <div
    class="classy-field__input-title"
  >
    <h4>
      End Time
    </h4>
  </div>
  <div>
    <div
      class="components-base-control classy-field__control classy-field__control--input classy-field__control--time-picker classy-highlight components-combobox-control emotion-0 emotion-1"
    >
      <div
        class="components-base-control__field emotion-2 emotion-3"
      >
        <div
          class="components-combobox-control__suggestions-container"
          tabindex="-1"
        >
          <div
            class="components-flex emotion-4 emotion-5 emotion-6"
            data-wp-c16t="true"
            data-wp-component="Flex"
          >
            <div
              class="components-flex-item components-flex-block emotion-7 emotion-6"
              data-wp-c16t="true"
              data-wp-component="FlexBlock"
            >
              <input
                aria-autocomplete="list"
                aria-describedby="components-form-token-suggestions-howto-combobox-control-10"
                aria-expanded="false"
                autocomplete="off"
                class="components-combobox-control__input components-form-token-field__input"
                id="components-form-token-input-combobox-control-10"
                role="combobox"
                size="8"
                type="text"
                value="1:00 pm"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
